common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java

```diff
@@ -37,6 +38,38 @@ public void validateCreateUserRequest(Request userRequest) {
     validateWebPages(userRequest);
   }
 
+  private void validateVerificationSourceAndCode(Request userRequest) {
+    String verificationCode = (String) userRequest.getRequest().get(JsonKey.VERIFICATION_CODE);
+    String verificationSource = (String) userRequest.getRequest().get(JsonKey.VERIFICATION_SOURCE);
+    if ((StringUtils.isBlank(verificationCode) && StringUtils.isBlank(verificationSource))
+        || (StringUtils.isNotBlank(verificationCode)
+            && StringUtils.isNotBlank(verificationSource))) {
+      if (StringUtils.isNotBlank(verificationSource)
+          && JsonKey.GOOGLE.equalsIgnoreCase(verificationSource)) {
+        validateParam(
+            (String) userRequest.getRequest().get(JsonKey.EMAIL),
+            ResponseCode.mandatoryParamsMissing,
+            JsonKey.EMAIL);
+      }
+    } else {
+      if (StringUtils.isBlank(verificationCode)) {
+        ProjectCommonException.throwClientErrorException(
```

Comment:
check if we cud replace with varargs -> shift formatting into the throwException method